great_expectations/datasource/new_datasource.py

```diff
@@ -32,6 +32,7 @@ def __init__(
         execution_engine=None,
```

Comment:
Using id was causing issues with instantiate_class_from_config. We may wish to refactor to use id in the future.